# Improved Medical Institute Matching Analysis #

## Set up ##

In [1]:
import pandas as pd
from tabulate import tabulate
from process import main

### Now to get the dataframe:

In [2]:
medical_institution = main("PubmedArticle.xml")
print(medical_institution)

100%|██████████| 3551/3551 [12:27<00:00,  4.75it/s]


      Article_PMID                                      Article_title  \
0         31644467  Common and rare forms of vasculitis associated...   
1         31642912  The quantitative assessment of interstitial lu...   
2         31642912  The quantitative assessment of interstitial lu...   
3         31642912  The quantitative assessment of interstitial lu...   
4         31642912  The quantitative assessment of interstitial lu...   
...            ...                                                ...   
23297     23477430  Primary anetoderma associated with primary Sjö...   
23298     23462883  DPP4 inhibitor-induced polyarthritis: a report...   
23299     23303389  Lymphoma risk in systemic lupus: effects of di...   
23300     23271426  Cryoglobulinemic vasculitis in systemic sclero...   
23301     23011161  Refractory thrombotic thrombocytopenic purpura...   

                                        Article_keywords  \
0                                                   None   
1  

### Of the all the rows in the table, how many of the Affiliation GRID name was matched successfully?

In [3]:
aff_value_df = medical_institution[medical_institution['Affiliation_name_PubMed'].notna()]

matched_count = aff_value_df["Affiliation_name_GRID"].value_counts().sum()
total = len(aff_value_df.index)
percentage_matched = (matched_count/total) * 100
print(percentage_matched)

53.17998455068235


The percentage of matched results have increased now from 11.3% previously to 53.2% .
Now we'll inspect those matches to check how accurate they are.

### Inspecting the successful Affiliation GRID matches

In [4]:

matches_found_df = aff_value_df[aff_value_df["Affiliation_name_GRID"].notna()]
df = matches_found_df[["Affiliation_name_PubMed", "Affiliation_name_GRID"]].head(20)
print(tabulate(df, headers='keys', tablefmt='psql'))

+----+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------+
|    | Affiliation_name_PubMed                                                                                                                                                                                                                                       | Affiliation_name_GRID                          |
|----+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------|
|  0 | Department of Pathophysiology, School of Medicine, Nation

Previous version of the process.py script could not catch organisations like 'National and Kapodistrian University of Athens' or 'Wuhan University'. 
However now from the table, this improved process.py script is very successful at finding and matching the organisations.

Below will show the random successful matches in the table.

In [6]:
df_random = matches_found_df[[
    "Affiliation_name_PubMed", "Affiliation_name_GRID"]].sample(10)
print(tabulate(df_random, headers='keys', tablefmt='psql'))

+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------+
|       | Affiliation_name_PubMed                                                                                                                                                                                                                                                                                                                                | Affiliation_name_GRID                |
|-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

### Inspecting the unsuccessful Affiliation GRID names 

In [10]:
no_grid_found_df = aff_value_df[aff_value_df["Affiliation_name_GRID"].isna()]
df = no_grid_found_df[["Affiliation_name_PubMed",
                       "Affiliation_name_GRID", "Affiliation_country", 'Affiliation_GRID_identifier']].head(20)
print(tabulate(df, headers='keys', tablefmt='psql'))

+----+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------+-----------------------+-------------------------------+
|    | Affiliation_name_PubMed                                                                                                                                                                                                               | Affiliation_name_GRID   | Affiliation_country   | Affiliation_GRID_identifier   |
|----+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------+-----------------------+-------------------------------|
|  1 | Department of Rheumatology, Am

In [9]:
df = no_grid_found_df[["Affiliation_name_PubMed",
                       "Affiliation_name_GRID", "Affiliation_country", 'Affiliation_GRID_identifier']].sample(10)
print(tabulate(df, headers='keys', tablefmt='psql'))

+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------------+-----------------------+-------------------------------+
|       | Affiliation_name_PubMed                                                                                                                                                                                                                                                                   | Affiliation_name_GRID   | Affiliation_country   | Affiliation_GRID_identifier   |
|-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

There seems to be multiple reasons for unsuccessful matches. 
Either the Affiliation_name_PubMed column value simply does not have the organisation/affiliation name in the text or the affiliation name is not in the institute.csv file. For example, for row 1 the Affiliation_name_PubMed is 'Department of Rheumatology, Amsterdam Rheumatology & Immunology Center. '. 
However for some values like row 52: 'Institute of Cellular Medicine, Newcastle University & NIHR Newcastle Biomedical Research Centre, Newcastle upon Tyne, NE2 4HH, UK.', the affiliation name is included in the institutes.csv but it's under the name 'NIHR Newcastle Biomedical Research Centre'. 
Since our matching score_cutoff is set to 0.9, this means the additional characters of 'Newcastle University &' has reduced the score significantly.

### Inspecting institutes.csv file

In [12]:
from os import environ
from dotenv import load_dotenv
from process import load_grid_institute_names

load_dotenv()
institutes = load_grid_institute_names(environ["GRID_INSTITUTE_FILEPATH"])
print(institutes[0])

Australian National University


Now lets have a look at some of the affiliation names in Affiliation_name_PubMed for those without a match.
For example, lets use row 52 which mentions Newcastle University & NIHR Newcastle Biomedical Research Centre in the column.
Let's check whether that unsuccessful match was because the institute is not in the csv file.

In [13]:
sample = 'Newcastle University & NIHR Newcastle Biomedical Research Centre'
is_institute_exist = [
    institute for institute in institutes if sample.lower() in institute.lower() or institute.lower() in sample.lower()]
print(is_institute_exist)

['Newcastle University', 'NIHR Newcastle Biomedical Research Centre', 'Search']


There are 2 different institutes excluding the 'Search'. 
There were 2 choices to match with but as for both choices, the  Affiliation_name_PubMed value contains the other characters which lowers the score of the match.

### Improvement

Overall, the matched institution results have improved from 11.3% to 53.2% and are very accurate due to the high score_cutoff of 0.9.
However, there are some matches that are missed within this process like institutions that contains additional words like University or other institutions. This lowers the score to an otherwise matched result so the next step to improve the matching further would be to split the phrases that are classified by the spacy nlp model as ORG or FAC into more accurate elements.
Another alternative approach is to switch the matching around so the institutes in the institute.csv list has to match the Affiliation_name_PubMed.